spec/realworld/edgecases_spec.rb

```diff
@@ -210,4 +210,21 @@
     bundle :install
     expect(exitstatus).to eq(0) if exitstatus
   end
+
+  context "when a gem inside a source block specifies a custom source" do
+    it "subsequent gems that are inside that source block are fetched from " \
+     "the correct source" do
+      install_gemfile <<-G, :expect_err => true
+        source "https://rubygems.org"
+
+        source 'https://rails-assets.org' do
+          gem 'rails-assets-angular', git: "git@github.com:sharoo/rails-assets-angular.git"
+          gem "i18n"
+        end
+      G
+
+      expect(out).to include("Could not find gem 'i18n' in rubygems repository https://rails-assets.org")
+      expect(exitstatus).to eq(7) if exitstatus
+    end
+  end
```

Comment:
This could also be tested inside `Dsl` but a realworld test is more suitable for such a bug imho. That said, I'm not happy with the fact that we depend on `rails-assets.org` being up *and* not containing `i18n`. Ideas?

cc @indirect @segiddins